src/main/compa/services/GsonService.java

```diff
@@ -1,41 +1,21 @@
 package main.compa.services;
 
 import com.google.gson.*;
-import main.compa.app.Adapter;
+import main.compa.app.Container;
 import main.compa.app.Service;
-import main.compa.models.Location;
 import org.bson.types.ObjectId;
 
 import java.lang.reflect.Type;
 
-public class GsonService implements Service {
-    GsonBuilder gb = new GsonBuilder();
+public class GsonService extends Service {
 
-    public GsonService() {
-        JsonSerializer<ObjectId> serializer = new JsonSerializer<ObjectId>() {
-            public JsonElement serialize(ObjectId src, Type typeOfSrc, JsonSerializationContext context) {
-                // #memories
-                //JsonObject jsonObj = new JsonObject();
-                //jsonObj.addProperty("value", src.toString());
+    Gson gson = new GsonBuilder().setPrettyPrinting().create();
 
-                return new JsonPrimitive(src.toString());//.parse(src.toString();
-            }
-        };
-                                                            // TODO: delete in prod
-        gb.registerTypeAdapter(ObjectId.class, serializer).setPrettyPrinting();
-        //gb.registerTypeAdapter(Location.class, new Adapter());
+    public GsonService(Container container) {
+        super(container);
     }
 
-    /** Instance unique pré-initialisée */
-    private static GsonService INSTANCE = new GsonService();
-
-    /** Point d'accès pour l'instance unique du singleton */
-    public static GsonService getInstance()
-    {
-        return INSTANCE;
-    }
-
-    public GsonBuilder getGsonBuilder(){
-        return this.gb;
+    public String toJson(Object thing){
+        return gson.toJson(thing);
     }
 }
```

Comment:
@ssssarah I don't like idea that every services are "container aware" 🧐

Reply:
you'll have access to the container inside the constructor, yes. But later on, it'll be stored in a private field of the superclass, so you can limit the service's access to the container by using limiting getters (that target only one resource of the container)